### PR TITLE
Create groups for ptokens

### PR DESCRIPTION
**Pull request by corydickson**
##### Description

Returns a 403 on the create ptoken endpoint if a user does not have the necessary permissions and hides the model the dashboard page. 

##### Testing

Run the `ptoken` migration to add a user to the group and create the necessary permissions. Then, login and navigate to the dashboard page. If a user has permission should see the modal, else it will not appear

### CODE REVIEW

File: app/ptokens/management/commands/setup_ptoken_launch.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        pToken_group_name = "pToken-Seed-Round"
+        pToken_group = Group.objects.get_or_create(name=pToken_group_name)[0]
+        ct = ContentType.objects.get_for_model(model=User)
+        add_pToken_auth = Permission.objects.get_or_create(name="Add pToken", codename="add_pToken_auth", content_type=ct)[0]
+        pToken_group.permissions.add(add_pToken_auth)
+
+        print('Adding seeded users to pToken Group')
+
+        # TODO: How else will we determine which users should be added to the group
+        valid_users = User.objects.filter(~Q(groups__name__in=[pToken_group_name]), profile__trust_profile=True)
```
Comment (corydickson) [inline app/ptokens/management/commands/setup_ptoken_launch.py:33]: Just need clarity on the identifier to seed the initial user base. Email / ids ?

File: app/ptokens/views.py [hunk 2]
```diff
@@ -204,6 +204,12 @@ def ptoken(request, token_id='me'):
             return JsonResponse(
                 {'error': _('You must be authenticated via github to use this feature!')},
                 status=401)
+
+        if not user.has_perm('auth.user.add_pToken_auth'):
```
Comment (corydickson) [inline app/ptokens/views.py:208]: @mds1 Brought up the good point that a user would be able to abuse Vue here and possibly open the modal; this ensures that they cannot make the request given their creds

**Review by aamustapha — APPROVED** [review]
LGTM

**Review by chibie — APPROVED** [review]

**Comment by HelloIPM** [comment]
OK NOTED

**Review by HelloIPM — APPROVED** [review]
DONE

File: app/ptokens/management/commands/setup_ptoken_launch.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        for user in valid_users:
+            user.groups.add(pToken_group)
+
+        print('Valid users added to pToken group')
```
Comment (thelostone-mc) [inline app/ptokens/management/commands/setup_ptoken_launch.py:37]: Do we run this just once and be done with it ?

Comment (corydickson) [reply app/ptokens/management/commands/setup_ptoken_launch.py:37]: yup, unless you had another way in mind. Let me know how seeding is determined and I can open another PR and add it to this command

**Review by thelostone-mc — APPROVED** [review]

**Comment by thelostone-mc** [comment]
Just tested this ! I ran `setup_ptoken_launch` and `has_ptoken_auth` is true but I don't see a creation modal popping up anywhere

Untitled

**Comment by mds1** [comment]
I'll defer to @corydickson on this one. From a quick glance [it looks like](https://github.com/corydickson/web/blob/96a5903de81cf3a919df7f443174b14a4bfd08ed/app/dashboard/templates/board/ptokens.html#L60-L73) it's set up correctly, where the "Create" button is behind `v-if="has_ptoken_auth"`.

Can we also add instructions/comments somewhere (even if just as a reply here) that explain what `setup_ptoken_launch` and `has_ptoken_auth` are both used for + how to configure them in dev mode? I've never used Django groups so a brief explanation would help!

**Pull request merged**